### DISCUSSION TITLE
Re-implementing `framework.py` and `model.py`'s API on top of Pydantic

### DISCUSSION THREAD

**Discussion by JGoldstone** [SMPTE/ris-osvp-metadata-camdkit#141] [Ideas]
I've written a not-quite-done-but-pretty-far-along re-implementation of the `Clip` class, and the various user-visible classes it uses, on top of Pydantic. A good starting point for what it can do [can be found here](https://docs.pydantic.dev/latest/concepts/models/). The work-in-progress implementation on top of Pydantic's `BaseModel` class can be seen [here](https://github.com/JGoldstone/pydantic-tests).

The title of that project, `pydantic-tests`, gives an indication of my uncertainty that the idea would actually pan out; but in fact I now think it could improve things (from a maintenance and extension perspective) by leveraging an external framework. In code for personal/consulting Python projects I anticipate I will be using Pydantic quite a bit.

**Comment by JGoldstone** [comment]
The work-in-progress implementation passes every unit test, matches each of the four examples ({complete, recommended} x {static, dynamic}), and produces an identical JSON schema to that which is produced by classic `camdkit` once the cleanup PR that I've submitted is merged. Currently that cleanup PR is blocked because of an AWS credentials error and I've asked James and Pierre for help getting past it.

There's a new file, `MIGRATION.md`, in the cleanup PR that explains the few changes that were required beyond cleanup; basically not sending the `subFrame` component of `TimecodeFormat` in a `Synchronization` block if it has the default value.

**Comment by jamesmosys** [comment]
This refactor to a pydantic model is now complete. Closing this.
